### PR TITLE
steps 1 and 2 of week-3-walkthrough

### PR DESCRIPTION
**Pull request by apluscs**

### CODE REVIEW

**Comment by acmcarther** [comment]
For future record:

We separated out the HTML changes from this change into branch "claire-html-format", then rebased this branch on top of that branch. 

This pull request needs to have it's base (merge into branch) changed back to master once claire-html-format is merged.

File: portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns a random greeting. */
+@WebServlet(urlPatterns = "/random-greeting")
+public final class RandomGreetingServlet extends HttpServlet {
+
+  private List<String> greetings;
```
Comment (acmcarther) [inline portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java:29]: If you have Google Guava already as a project dependency, then ImmutableList is a slightly better fit than List or ArrayList here. You'll know if you do or not by including the `import` and then recompiling to make sure that works. If you don't have it yet, then skip this comment for now... I think adding dependencies to maven is a separate unit.

```java
import com.google.common.collect.ImmutableList;
```

```java
  private ImmutableList<String> greetings;
```

```java
  @Override
  public void init() {
    greetings = ImmutableList.of(
        "Hello!",
        "Greetings!",
        "Welcome!",
        "How do you do?",
        "Hi!");
  }
```

File: portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    greetings = new ArrayList<>();
+    greetings.add("Hello!");
+    greetings.add("Greetings!");
+    greetings.add("Welcome!");
+    greetings.add("How do you do?");
+    greetings.add("Hi!");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
[the remark this answers is not in the document]
Comment (apluscs) [reply portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java:42]: l'll drop it, l just copy pasted this part from their tutorial, in which they never threw an IOException but kept in the header

File: portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java (hunk 1 above)
Comment (apluscs) [reply portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java:29]: Does not compile :(
package com.google.common.collect does not exist

File: portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (acmcarther) [reply portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java:42]: Acknowledged, resolving.

File: portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java (hunk 1 above)
Comment (acmcarther) [reply portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java:29]: Acknowledged, resolving

File: portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (acmcarther) [reply portfolio/src/main/java/com/google/sps/servlets/RandomGreetingServlet.java:42]: Ack, resolving.

**Pull request merged**